### PR TITLE
roachpb: hold all measurements within a timeseries sample

### LINKED ISSUE #5518: storage: Separate out consistency checks over floating point data
We run periodic consistency checks over all the replicated data. Timeseries data which is cockroach internal, is replicated consistently, but merged independently often outside of raft on each replica. While the consistency checker forces a merge on each replica at a snapshot, thereby making it consistent before computing a checksum on the data, the data itself contains floating points, with the merge operator running floating point math like adding up all the timeseries values. We'd like to continue running consistency checks on this data so we can catch consistency problems on homogenous clusters, while accepting consistency differences on heterogenous one.

Options:
1. Separate out the checksum over timeseries data so that it doesn't interfere with the range checksum (preferred).
2. Convert the floating point values into integers and run integer math on them. Not recommended because we also really don't care about this data being that accurate.

### PR DESCRIPTION
**Pull request by vivekmenezes**
closes #5518 

I can make this change to close out this issue as long as we agree to eventually land it.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5836)

### CODE REVIEW

File: roachpb/internal.proto [hunk 1]
```diff
@@ -75,12 +73,13 @@ message InternalTimeSeriesSample {
...
-  optional double sum = 7 [(gogoproto.nullable) = false];
+  // A series of measurements gathered over a time period.
+  repeated double value = 2;
   // Maximum encountered measurement in this sample.
   optional double max = 8;
   // Minimum encountered measurement in this sample.
   optional double min = 9;
+
+  optional uint32 DELETED_count = 6 [(gogoproto.nullable) = false];
+  optional double DELETED_sum = 7 [(gogoproto.nullable) = false];
```
Comment (petermattis) [inline roachpb/internal.proto]: I'm not sure you can mark these fields as `DELETED` just yet. Won't new servers need to read these fields if they exist?

File: roachpb/internal.proto [hunk 2]
```diff
@@ -75,12 +73,13 @@ message InternalTimeSeriesSample {
   // the TimeSeriesData collection.
   optional int32 offset = 1 [(gogoproto.nullable) = false];
 
-  // Count of measurements taken within this sample.
-  optional uint32 count = 6 [(gogoproto.nullable) = false];
-  // Sum of all measurements.
-  optional double sum = 7 [(gogoproto.nullable) = false];
+  // A series of measurements gathered over a time period.
+  repeated double value = 2;
```
Comment (petermattis) [inline roachpb/internal.proto]: This looks fine to me.

**Comment by vivekmenezes** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[roachpb/internal.proto, line 84 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5836#-KEWZTJG5o1bN2EXk_b--r1-84:-KEWn683_-4_Fxd4wVq4:1548514410) ([raw file](https://github.com/cockroachdb/cockroach/blob/d617dcbe1e3e4d678a5133e6bd9d5d1b1fb0648b/roachpb/internal.proto#L84)):*
That was my intention. I forget that we only mark fields as DELETED once code doesn't refer to them.

I realize now that this change needs to be rolled out in a number of phases.
1. Code that knows how to process the new field "value"

once that has been rolled out everywhere
2. Code that generates the new field "value"

and once that's rolled out everywhere, start cleanup process

So, I'm definitely not inclined to want to make this change :-)

Another option is to eliminated ts data from the consistency check temporarily.
Not a pretty option too :-)

We should not do this change at all, and just switch this mechanism to CPut. Merge is just a bad idea.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5836)*

**Comment by petermattis** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[roachpb/internal.proto, line 84 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5836#-KEWZTJG5o1bN2EXk_b--r1-84:-KEWrZ9ULlT2e0ttw_Wu:-1012970863) ([raw file](https://github.com/cockroachdb/cockroach/blob/d617dcbe1e3e4d678a5133e6bd9d5d1b1fb0648b/roachpb/internal.proto#L84)):*
Another option is to change the keys under which time series data is stored, effectively tossing out data stored under the old keys. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5836)*

**Comment by vivekmenezes** [comment]
updating the issue asking mtracy for his input on whether we should just
move to CPut. That would be nicer

**Comment by mrtracy** [comment]
This is no longer necessary, #5905 will address the underlying concerns in a different fashion.

**Pull request closed without merging**